rclcpp/include/rclcpp/node.hpp

```diff
@@ -720,6 +720,25 @@ class Node : public std::enable_shared_from_this<Node>
     ParameterT & parameter,
     const ParameterT & alternative_value) const;
 
+  /// Get the parameter value, or the "alternative_value" if not set, and assign it to "parameter".
+  /**
+   * If the parameter was not set, then the "parameter" argument is assigned
+   * the "alternative_value".
+   *
+   * This method will not throw the rclcpp::exceptions::ParameterNotDeclaredException exception.
+   *
+   * In all cases, the parameter is never set or declared within the node.
+   *
+   * \param[in] name The name of the parameter to get.
+   * \param[in] alternative_value Value to be stored in output if the parameter was not set.
+   * \returns The value of the parameter.
+   */
+  template<typename ParameterT>
```

Comment:
Right?
```suggestion
  RCLCPP_PUBLIC
  template<typename ParameterT>
```

Reply:
I'm not sure, I just copied from `get_parameter_or(const std::string & name, ParameterT & parameter, const ParameterT & alternative_value)`.
Will wait for comments from maintainers.

Reply:
I think this is intentionally NOT `RCLCPP_PUBLIC` to reduce loading time for DSO, if i am not mistaken. So i think we do not need that here either.

even if we do that, i think the following is the correct suggestion.

```suggestion
  template<typename ParameterT>
  RCLCPP_PUBLIC
```

Reply:
Sorry, I misunderstood. `RCLCPP_PUBLIC` is not required because you can't really export a template - it has to be instantiated, which is why the templated functions are defined in `node_impl.hpp` and thus not `dllexport`-ed.